# Titanic Dataset

First, the data is loaded and some basic information is printed to get an idea about the data set.

In [15]:
import numpy as np
import pandas as pd
#import csv

data = pd.read_csv("titanic-data.csv")

print(sum(data["Survived"]), "of 891 survived")
data.describe()


342 of 891 survived


PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

I will analyse the Titanic dataset to find factors which increase the chance to survive the tragedy and those which did not have an effect.

Therefore I define the "chance of survival" as the share of survivors in the population or a subgroup thereof. For example, over the whole population the chance to survive is 342/891 = 0.383838.

Lets compare  how different factors influence this chance. In the following, the data is grouped by sex, ticket class and the port of embarkation.

In [37]:
data.groupby("Pclass").first()

PassengerId  Survived  \
Pclass                          
1                 2         1   
2                10         1   
3                 1         0   

                                                     Name     Sex   Age  \
Pclass                                                                    
1       Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0   
2                     Nasser, Mrs. Nicholas (Adele Achem)  female  14.0   
3                                 Braund, Mr. Owen Harris    male  22.0   

        SibSp  Parch     Ticket     Fare Cabin Embarked  
Pclass                                                   
1           1      0   PC 17599  71.2833   C85        C  
2           1      0     237736  30.0708   D56        C  
3           1      0  A/5 21171   7.2500    G6        S